docs/network-setup.md

```diff
@@ -76,10 +76,6 @@ to your configuration file like this:
 Alternatively, if you are using firectl, add
 --tap-device=tap0/AA:FC:00:00:00:01` to your command line.
 
-*Note:* When `guest_mac` is not explicitely specified, a MAC address is generated
-and assigned to the network interface. This address is included in the result of
-GET `/vm/config`.
-
```

Comment:
Do we need to add any information regarding what happens when the user does not specify a MAC address?

```
*Note:* When `guest_mac` is not explicitly specified, a MAC address is generated
by the driver and assigned to the network interface. This address cannot be
queried via Firecracker
```

WDYT @dianpopa  @pb8o ?

Reply:
@vibharya you mean let them know that the driver will assign one for them?

As far as the CHANGELOG, yes, we need to remove the entries you added.

Reply:
Yes, let them know one will be assigned for them. And that we cannot retrieve/report it using FC.

Yes, we can remove the entries I added and instead mention that they are now reported as null (or "None" or excluded - not sure exactly what/vm/config is returning now in this case) pre and post snapshot resume.
Hence fixing [this part](https://github.com/firecracker-microvm/firecracker/issues/2943#:~:text=net%20devices%20MAC%20addresses%20are%20reported%20as%20all%20zeros%20if%20left%20unconfigured.%20View%20Partial%20fix%20for%20restore%20snapshot%20GET%20/vm/config%20bug%C2%A0%232806%20(comment)) of the issue.

Reply:
Btw, can't we retrieve the MAC with `write_config`?

Reply:
@dianpopa @vibharya I changed `CHANGELOG`. Your thoughts?

Reply:
sounds good to me

Reply:
Have you been able to retrieve the MAC address using `write_config`?

Also, what do we get when the MAC address is left unspecified currently?

Reply:
When the MAC address is left unspecified, we get back `None`. Exactly what the user configured.

In oder to receive the MAC_ADDR from the driver through `write_config`, we need to negotiate the `VIRTIO_NET_F_CTRL_MAC_ADDR` which needs to have a control queue set up. However, since our initial intent with the `GetConfig` was to give back the config provided by the user, we decided that obtaining the actual mac addr set by the driver is not part of that feature. Also, if the user has a real need to know the mac address from inside the microVM, they always have the possibility to specify one.

Reply:
Just verified that the output of GET /vm/config (even for boot from snapshot) would show the network interface as something like
`{
      "iface_id":"1",
      "host_dev_name":"vmtap30",
      "guest_mac":null,
      "rx_rate_limiter":null,
      "tx_rate_limiter":null
}`
Thats good, was the case previously as well.


> When the MAC address is left unspecified, we get back `None`. Exactly what the user configured.
> 
> In oder to receive the MAC_ADDR from the driver through `write_config`, we need to negotiate the `VIRTIO_NET_F_CTRL_MAC_ADDR` which needs to have a control queue set up. However, since our initial intent with the `GetConfig` was to give back the config provided by the user, we decided that obtaining the actual mac addr set by the driver is not part of that feature. Also, if the user has a real need to know the mac address from inside the microVM, they always have the possibility to specify one.

Yes, I understand and agree with this part 100%.
I'm only wondering if we need a note in the snapshot_support doc regarding this or not.